src/azul/deployment.py

```diff
@@ -297,5 +301,71 @@ def _sanitize_tf(tf_config: JSON) -> JSON:
     return {k: v for k, v in tf_config.items() if v}
 
 
+def _fix_name_tag(tf_config: JSON) -> JSON:
+    """
+    Renames the ``name`` tag generated by :func:`get_cloud_tags` and uppercases
+    it to ``Name`` for AWS resources with tags.
+
+    >>> resource = {'resource_name': {'tags': {'name': 'foo'}}}
+    >>> fixed = _fix_name_tag({'resource': [{'aws_resource_type': resource}]})
+    >>> tags = fixed['resource'][0]['aws_resource_type']['resource_name']['tags']
+    >>> assert 'Name' in tags
+    >>> assert 'name' not in tags
+    """
+    for group in filter(lambda x: isinstance(x, dict), tf_config.get('resource', [])):
```

Comment:
Use the terminology from https://www.terraform.io/docs/configuration/syntax-json.html and be sure you handle [nested blocks](https://www.terraform.io/docs/configuration/syntax-json.html#nested-block-mapping)